ql/src/java/org/apache/hadoop/hive/ql/queryhistory/repository/IcebergRepository.java

```diff
@@ -95,6 +96,7 @@ protected Table createTable(Hive hive, Database db) throws HiveException {
     table.setProperty(hive_metastoreConstants.META_TABLE_NAME, QUERY_HISTORY_DB_TABLE_NAME);
 
     table.setFields(schema.getFields());
+    table.setProperty(ICEBERG_METADATA_CLEANUP_CONFIG, "false");
```

Comment:
isn't it a catalog property? https://github.com/apache/hive/pull/6218/files#diff-427b2ffcf14c9599990dd80ea356f9fabbca2fae484f47ee8a3637263b2fa79eR240

Reply:
This property can be defined as a table property or on the session level.
In `BaseHiveIcebergMetaHook.preCreateTable()` it retrieves this prop from table properties or session and sets it in the catalog properties. This is the same behavior as for the other iceberg properties.